### PR TITLE
Implement ActivityGroupScreen

### LINKED ISSUE #337: Implement ActivityGroupScreen
## Goal
Create a new screen (`ActivityGroupScreen.kt`) that displays all **activities** of a group — both *events* and *series* — using data exposed by `ActivityGroupViewModel`.

### 1. Create `ActivityGroupScreen.kt`
The screen must:

- Accept a `groupId: String` as a parameter.
- Fetch UI state from `ActivityGroupViewModel`, which provides:
  ```kotlin
  uiState.events  // List<Event>
  uiState.series  // List<Serie>
  uiState.isLoading
  uiState.error
  ```

### 2. Load activities via ViewModel
The ViewModel already handles backend calls:
```kotlin
eventsRepository.getEventsByIds(group.eventIds)
seriesRepository.getSeriesByIds(group.serieIds)
```

### 3. Render the activities
UI requirements:
- Show all activities in a scrollable list.
- No separation between *ongoing* and *upcoming* (different from `OverviewScreen`).
- Reuse layout patterns from `OverviewScreen` (cards, spacing, typography).

### 4. Handle all UI states
The screen must support:
- **Loading state** → loading indicator
- **Error state** → error message or layout
- **Empty state** → message like  
  *"This group has no activities yet"*
- **Success state** → list of events + series

## Figma Screenshots:

<p align="center">
  ActivityGroupScreenEmpty
  ActivityGroupScreenFilled
</p>

### PR DESCRIPTION
**Pull request by BryanDMP**
#  Implement ActivityGroupScreen

Closes #337

##  Overview

This PR introduces `ActivityGroupScreen`, a new screen that displays all activities (events and series) associated with a specific group. The screen provides a unified view of group activities without categorization into ongoing/upcoming sections, following the design specifications from Figma.

### 1. **ActivityGroupScreen.kt**
A new composable screen that:
- Accepts a `groupId` parameter to load group-specific activities
- Observes `ActivityGroupViewModel.uiState` for data and loading states
- Displays events and series in a single scrollable list
- Includes back navigation with arrow icon in top bar
- Handles all UI states: loading, empty, content, and error

### 2. **ActivityGroupScreenTest.kt**
Comprehensive test coverage with 11 optimized Robolectric tests:
- UI display and navigation tests (top bar, back button)
- Content rendering tests (events, series, mixed content)
- User interaction tests (card clicks, callbacks)
- Edge cases (empty state, events-only, series-only)
- Error handling validation

##  Design Implementation

The screen was mostly inspired by the existing design system from `OverviewScreen`:
- Uses `EventCard` and `SerieCard` components for consistent UI
- Applies `MaterialTheme` semantic colors and typography
- Maintains standard spacing with `Dimens` constants
- Includes proper accessibility with test tags

---

**Related PRs:** Navigation integration will follow in subsequent PR

### CODE REVIEW

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+
+    composeTestRule.setContent {
+      ActivityGroupScreen(
+          groupId = "1",
+          activityGroupViewModel = viewModel,
+          onNavigateBack = { backClicked = true })
+    }
+
+    composeTestRule.waitForIdle()
+    composeTestRule.mainClock.advanceTimeBy(2000)
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: Hardcoded 2000ms delay appears in every test. Should be a constant.

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+            groupRepository = groupRepo, eventsRepository = eventRepo, seriesRepository = serieRepo)
+    var selectedEvent: Event? = null
+
+    composeTestRule.setContent {
+      ActivityGroupScreen(
+          groupId = "1", activityGroupViewModel = viewModel, onSelectEvent = { selectedEvent = it })
+    }
+
+    composeTestRule.waitForIdle()
+    composeTestRule.mainClock.advanceTimeBy(2000)
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: Hardcoded 2000ms delay appears in every test. Should be a constant.

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+    val viewModel =
+        ActivityGroupViewModel(
+            groupRepository = groupRepo, eventsRepository = eventRepo, seriesRepository = serieRepo)
+
+    composeTestRule.setContent {
+      ActivityGroupScreen(groupId = "1", activityGroupViewModel = viewModel)
+    }
+
+    composeTestRule.waitForIdle()
+    composeTestRule.mainClock.advanceTimeBy(2000)
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: Hardcoded 2000ms delay appears in every test. Should be a constant.

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+  }
+
+  @Test
+  fun activityGroupScreen_backButtonTriggersCallback() {
+    val groupRepo = FakeGroupRepository()
+    val eventRepo = FakeActivityGroupEventsRepository()
+    val serieRepo = FakeActivityGroupSeriesRepository()
+    val viewModel =
+        ActivityGroupViewModel(
+            groupRepository = groupRepo, eventsRepository = eventRepo, seriesRepository = serieRepo)
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: This setup code is duplicated across the entire test file. To make it cleaner and reduce redundancy, I would extract it into a `@Before` method, for example:
```
private lateinit var groupRepo: FakeGroupRepository
private lateinit var eventRepo: FakeActivityGroupEventsRepository
private lateinit var serieRepo: FakeActivityGroupSeriesRepository
private lateinit var viewModel: ActivityGroupViewModel
@Before
fun setUp() {
  groupRepo = FakeGroupRepository()
  eventRepo = FakeActivityGroupEventsRepository()
  serieRepo = FakeActivityGroupSeriesRepository()
  viewModel = ActivityGroupViewModel(
    groupRepository = groupRepo,
    eventsRepository = eventRepo,
    seriesRepository = serieRepo
  )
}
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 5]
```diff
@@ -0,0 +1,502 @@
...
+    composeTestRule.setContent {
+      ActivityGroupScreen(
+          groupId = "1",
+          activityGroupViewModel = viewModel,
+          onNavigateBack = { backClicked = true })
+    }
+
+    composeTestRule.waitForIdle()
+    composeTestRule.mainClock.advanceTimeBy(2000)
+    composeTestRule.waitForIdle()
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: This setup code is repeated throughout the test file. To simplify and avoid duplication, you can extract it into a dedicated helper method, for example:
```
private fun waitForContent() {
  composeTestRule.waitForIdle()
  composeTestRule.mainClock.advanceTimeBy(2000)
  composeTestRule.waitForIdle()
}
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 6]
```diff
@@ -0,0 +1,502 @@
...
+  }
+
+  @Test
+  fun activityGroupScreen_emptyState_displaysCorrectMessage() {
+    val groupRepo = FakeGroupRepository()
+    val eventRepo = FakeActivityGroupEventsRepository()
+    val serieRepo = FakeActivityGroupSeriesRepository()
+    val viewModel =
+        ActivityGroupViewModel(
+            groupRepository = groupRepo, eventsRepository = eventRepo, seriesRepository = serieRepo)
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: This setup code is duplicated across the entire test file. To make it cleaner and reduce redundancy, I would extract it into a `@Before` method, for example:
```
private lateinit var groupRepo: FakeGroupRepository
private lateinit var eventRepo: FakeActivityGroupEventsRepository
private lateinit var serieRepo: FakeActivityGroupSeriesRepository
private lateinit var viewModel: ActivityGroupViewModel
@Before
fun setUp() {
  groupRepo = FakeGroupRepository()
  eventRepo = FakeActivityGroupEventsRepository()
  serieRepo = FakeActivityGroupSeriesRepository()
  viewModel = ActivityGroupViewModel(
    groupRepository = groupRepo,
    eventsRepository = eventRepo,
    seriesRepository = serieRepo
  )
}
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 7]
```diff
@@ -0,0 +1,502 @@
...
+
+    runBlocking { groupRepo.addGroup(Group(id = "1", name = "Empty Group")) }
+    composeTestRule.setContent {
+      ActivityGroupScreen(groupId = "1", activityGroupViewModel = viewModel)
+    }
+
+    composeTestRule.waitForIdle()
+    composeTestRule.mainClock.advanceTimeBy(2000)
+    composeTestRule.waitForIdle()
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: This setup code is repeated throughout the test file. To simplify and avoid duplication, you can extract it into a dedicated helper method, for example:
```
private fun waitForContent() {
  composeTestRule.waitForIdle()
  composeTestRule.mainClock.advanceTimeBy(2000)
  composeTestRule.waitForIdle()
}
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 8]
```diff
@@ -0,0 +1,502 @@
...
+    var selectedEvent: Event? = null
+
+    composeTestRule.setContent {
+      ActivityGroupScreen(
+          groupId = "1", activityGroupViewModel = viewModel, onSelectEvent = { selectedEvent = it })
+    }
+
+    composeTestRule.waitForIdle()
+    composeTestRule.mainClock.advanceTimeBy(2000)
+    composeTestRule.waitForIdle()
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: This setup code is repeated throughout the test file. To simplify and avoid duplication, you can extract it into a dedicated helper method, for example:

```
private fun waitForContent() {
  composeTestRule.waitForIdle()
  composeTestRule.mainClock.advanceTimeBy(2000)
  composeTestRule.waitForIdle()
}
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 9]
```diff
@@ -0,0 +1,502 @@
...
+
+    assert(selectedSerie != null)
+    assert(selectedSerie?.serieId == "serie1")
+  }
+
+  @Test
+  fun activityGroupScreen_hasUniqueTestTags() {
+    val groupRepo = FakeGroupRepository()
+    val eventRepo = FakeActivityGroupEventsRepository()
+    val serieRepo = FakeActivityGroupSeriesRepository()
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: This setup code is duplicated across the entire test file. To make it cleaner and reduce redundancy, I would extract it into a `@Before` method, for example:

```
private lateinit var groupRepo: FakeGroupRepository
private lateinit var eventRepo: FakeActivityGroupEventsRepository
private lateinit var serieRepo: FakeActivityGroupSeriesRepository
@Before
fun setUp() {
  groupRepo = FakeGroupRepository()
  eventRepo = FakeActivityGroupEventsRepository()
  serieRepo = FakeActivityGroupSeriesRepository()
}
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 10]
```diff
@@ -0,0 +1,502 @@
...
+        ActivityGroupViewModel(
+            groupRepository = groupRepo, eventsRepository = eventRepo, seriesRepository = serieRepo)
+
+    composeTestRule.setContent {
+      ActivityGroupScreen(groupId = "1", activityGroupViewModel = viewModel)
+    }
+
+    composeTestRule.waitForIdle()
+    composeTestRule.mainClock.advanceTimeBy(2000)
+    composeTestRule.waitForIdle()
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: This setup code is repeated throughout the test file. To simplify and avoid duplication, you can extract it into a dedicated helper method, for example:

```
private fun waitForContent() {
  composeTestRule.waitForIdle()
  composeTestRule.mainClock.advanceTimeBy(2000)
  composeTestRule.waitForIdle()
}
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 11]
```diff
@@ -0,0 +1,502 @@
...
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: This setup code is repeated throughout the test file. To simplify and avoid duplication, you can extract it into a dedicated helper method, for example:

```
private fun waitForContent() {
  composeTestRule.waitForIdle()
  composeTestRule.mainClock.advanceTimeBy(2000)
  composeTestRule.waitForIdle()
}
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 12]
```diff
@@ -0,0 +1,502 @@
...
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: This setup code is repeated throughout the test file. To simplify and avoid duplication, you can extract it into a dedicated helper method, for example:

```
private fun waitForContent() {
  composeTestRule.waitForIdle()
  composeTestRule.mainClock.advanceTimeBy(2000)
  composeTestRule.waitForIdle()
}
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 13]
```diff
@@ -0,0 +1,502 @@
...
+
+    composeTestRule.onNodeWithText("Event Only 1").assertIsDisplayed()
+    composeTestRule.onNodeWithText("Event Only 2").assertIsDisplayed()
+  }
+
+  @Test
+  fun activityGroupScreen_groupWithOnlySeries_displaysCorrectly() {
+    val groupRepo = FakeGroupRepository()
+    val eventRepo = FakeActivityGroupEventsRepository()
+    val serieRepo = FakeActivityGroupSeriesRepository()
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: This setup code is duplicated across the entire test file. To make it cleaner and reduce redundancy, I would extract it into a `@Before` method, for example:

```
private lateinit var groupRepo: FakeGroupRepository
private lateinit var eventRepo: FakeActivityGroupEventsRepository
private lateinit var serieRepo: FakeActivityGroupSeriesRepository
@Before
fun setUp() {
  groupRepo = FakeGroupRepository()
  eventRepo = FakeActivityGroupEventsRepository()
  serieRepo = FakeActivityGroupSeriesRepository()
}
```

**Review by Atilla13 — CHANGES_REQUESTED** [review]
Excellent work, @BryanDMP! This is a major milestone for the project. Displaying all activities (events and series) for a group is a core feature. I've tested your implementation locally and it works perfectly!

I have a couple of suggestions to improve the test file's maintainability and reduce CI running time:

**1. Remove redundant tests**

`activityGroupScreen_groupWithOnlyEvents_displaysCorrectly()` and `activityGroupScreen_groupWithOnlySeries_displaysCorrectly()` can be removed since `activityGroupScreen_displaysEventsAndSeriesInSingleList()` already covers mixed content scenarios.

**2. Extract repeated setup code**

Most tests repeat this pattern:
```kotlin
val groupRepo = FakeGroupRepository()
val eventRepo = FakeActivityGroupEventsRepository()
val serieRepo = FakeActivityGroupSeriesRepository()
// ... setup ...
composeTestRule.waitForIdle()
composeTestRule.mainClock.advanceTimeBy(2000)
composeTestRule.waitForIdle()
```

Consider extracting this to a helper function or `@Before` setup method to reduce duplication.

These changes will keep the test suite clean and efficient. I'll approve once you've made these adjustments. Thanks for the great work!

**Comment by BryanDMP** [comment]
I appreciate the requested changes and the focus on tests is important since we discussed it at the last scrum meeting. If the requested changes are accepted, I'll move my PR about the navigation for this screen from draft to Pull Request ready. Thanks @Atilla13 !

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 14]
```diff
@@ -0,0 +1,417 @@
...
+          activityGroupViewModel = viewModel,
+          onSelectedSerie = { selectedSerie = it })
+    }
+
+    waitForContent()
+
+    composeTestRule.onNodeWithText("Clickable Serie").performClick()
+
+    assert(selectedSerie != null)
+    assert(selectedSerie?.serieId == "serie1")
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: Replace with:
```
assertNotNull(selectedSerie)
assertEquals("serie1", selectedSerie?.serieId)
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 15]
```diff
@@ -0,0 +1,417 @@
...
+    composeTestRule.onNodeWithText("Group Activities").assertIsDisplayed()
+    composeTestRule.onNodeWithTag(ActivityGroupScreenTestTags.BACK_BUTTON).assertIsDisplayed()
+    composeTestRule.onNodeWithContentDescription("Back").assertIsDisplayed()
+  }
+
+  @Test
+  fun activityGroupScreen_backButtonTriggersCallback() {
+    val viewModel =
+        ActivityGroupViewModel(
+            groupRepository = groupRepo, eventsRepository = eventRepo, seriesRepository = serieRepo)
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: You could create a helper function instead of a shared instance:

```kotlin
private fun createViewModel(
    groupRepo: GroupRepository = this.groupRepo,
    eventRepo: EventsRepository = this.eventRepo,
    serieRepo: SeriesRepository = this.serieRepo
) = ActivityGroupViewModel(
    groupRepository = groupRepo,
    eventsRepository = eventRepo,
    seriesRepository = serieRepo
)
```

Then here you can simply write:

```kotlin
val viewModel = createViewModel()
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 16]
```diff
@@ -0,0 +1,417 @@
...
+    composeTestRule.onNodeWithTag(ActivityGroupScreenTestTags.BACK_BUTTON).performClick()
+
+    assert(backClicked)
+  }
+
+  @Test
+  fun activityGroupScreen_emptyState_displaysCorrectMessage() {
+    val viewModel =
+        ActivityGroupViewModel(
+            groupRepository = groupRepo, eventsRepository = eventRepo, seriesRepository = serieRepo)
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: You could create a helper function instead of a shared instance:

```kotlin
private fun createViewModel(
    groupRepo: GroupRepository = this.groupRepo,
    eventRepo: EventsRepository = this.eventRepo,
    serieRepo: SeriesRepository = this.serieRepo
) = ActivityGroupViewModel(
    groupRepository = groupRepo,
    eventsRepository = eventRepo,
    seriesRepository = serieRepo
)
```

Then here you can simply write:

```kotlin
val viewModel = createViewModel()
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 17]
```diff
@@ -0,0 +1,417 @@
...
+          Group(
+              id = "1",
+              name = "Active Group",
+              eventIds = listOf("event1", "event2"),
+              serieIds = listOf("serie1")))
+    }
+
+    val viewModel =
+        ActivityGroupViewModel(
+            groupRepository = groupRepo, eventsRepository = eventRepo, seriesRepository = serieRepo)
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: You could create a helper function instead of a shared instance:

```kotlin
private fun createViewModel(
    groupRepo: GroupRepository = this.groupRepo,
    eventRepo: EventsRepository = this.eventRepo,
    serieRepo: SeriesRepository = this.serieRepo
) = ActivityGroupViewModel(
    groupRepository = groupRepo,
    eventsRepository = eventRepo,
    seriesRepository = serieRepo
)
```

Then here you can simply write:

```kotlin
val viewModel = createViewModel()
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 18]
```diff
@@ -0,0 +1,417 @@
...
+    val serie = createTestSerie("serie1", "Clickable Serie")
+
+    runBlocking {
+      serieRepo.addSerie(serie)
+      groupRepo.addGroup(Group(id = "1", name = "Test Group", serieIds = listOf("serie1")))
+    }
+
+    val viewModel =
+        ActivityGroupViewModel(
+            groupRepository = groupRepo, eventsRepository = eventRepo, seriesRepository = serieRepo)
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: You could create a helper function instead of a shared instance:

```kotlin
private fun createViewModel(
    groupRepo: GroupRepository = this.groupRepo,
    eventRepo: EventsRepository = this.eventRepo,
    serieRepo: SeriesRepository = this.serieRepo
) = ActivityGroupViewModel(
    groupRepository = groupRepo,
    eventsRepository = eventRepo,
    seriesRepository = serieRepo
)
```

Then here you can simply write:

```kotlin
val viewModel = createViewModel()
```

File: app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt [hunk 19]
```diff
@@ -0,0 +1,417 @@
...
+  }
+
+  @Test
+  fun activityGroupScreen_handlesErrorFromRepository() {
+    val errorGroupRepo = FakeGroupRepository(shouldThrowError = true)
+    val viewModel =
+        ActivityGroupViewModel(
+            groupRepository = errorGroupRepo,
+            eventsRepository = eventRepo,
+            seriesRepository = serieRepo)
```
Comment (Atilla13) [inline app/src/test/java/com/android/joinme/ui/groups/ActivityGroupScreenTest.kt]: You could create a helper function instead of a shared instance:

```kotlin
private fun createViewModel(
    groupRepo: GroupRepository = this.groupRepo,
    eventRepo: EventsRepository = this.eventRepo,
    serieRepo: SeriesRepository = this.serieRepo
) = ActivityGroupViewModel(
    groupRepository = groupRepo,
    eventsRepository = eventRepo,
    seriesRepository = serieRepo
)
```

Then here you can simply write:

```kotlin
val viewModel = createViewModel(groupRepo = FakeGroupRepository(shouldThrowError = true))
```

**Comment by BryanDMP** [comment]
Look, this is the second time you're coming back with feedback on the same file. The assert() vs assertTrue() thing, the ViewModel helper,...These aren't subtle edge cases. They're basic stuff that should've been in your first review.
I get that reviews take time, but when you're splitting feedback across two rounds on one test file, you're just wasting everyone's time. 
Do a proper pass the first time. Catch everything at once. That's it.
I'll fix it now, but let's not make this a pattern...

**Comment by Atilla13** [comment]
Quote:
> I take responsibility for not catching them earlier, and I appreciate your understanding.
> I’ll make sure to be more attentive in my future reviews.

**Review by Atilla13 — APPROVED** [review]
LGTM! 
Ready to be merged into main!
Thank you for your time @BryanDMP.
